hapic/doc.py

```diff
@@ -261,5 +262,23 @@ def save_in_file(
 
 
 # TODO BS 20171109: Must take care of already existing definition names
-def generate_schema_name(schema):
-    return schema.__name__
+def generate_schema_name(schema: marshmallow.Schema):
+    """
+    Return best candidate name for one schema cls or instance.
+    :param schema: instance or cls schema
+    :return: best schema name
+    """
+    if not isinstance(schema, type):
+        schema = type(schema)
+
+    if getattr(schema, '_schema_name', None):
+        if schema.opts.exclude:
```

Comment:
you do not manage "only" case, do you?

Reply:
yes but only case are handled the same way as only, only is a specific case of exclude.